buildSrc/src/main/java/org/elasticsearch/gradle/Distribution.java

```diff
@@ -20,23 +20,66 @@
 
 public enum Distribution {
 
-    INTEG_TEST("integ-test", "zip"),
-    ZIP("elasticsearch", "zip"),
-    ZIP_OSS("elasticsearch-oss", "zip");
+    INTEG_TEST("integ-test"),
+    DEFAULT("elasticsearch"),
+    OSS("elasticsearch-oss");
 
     private final String fileName;
-    private final String fileExtension;
 
-    Distribution(String name, String fileExtension) {
+    Distribution(String name) {
         this.fileName = name;
-        this.fileExtension = fileExtension;
     }
 
-    public String getFileName() {
+    public String getArtifactName() {
         return fileName;
     }
 
     public String getFileExtension() {
-        return fileExtension;
+        if (this.equals(INTEG_TEST)) {
+            return "zip";
+        } else {
+            switch (getOS()) {
+                case LINUX:
```

Comment:
Personally, I always like to add an explicit comment whenever there is a purposeful fall-through in a `switch` expression, just so future folks know it wasn't a mistake.

Reply:
I ended up removing this switch statement.